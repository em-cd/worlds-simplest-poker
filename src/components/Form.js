import React from 'react';

function Form(props) {
  return (
    <form onSubmit={(event) => props.onSubmit(event)}>
      <label>
        How many players?
        <input
          id='players'
          type="number"
          name="inputPlayers"
          value={props.inputPlayers}
          min="2"
          max="52"
          onChange={(event) => props.onChange(event)} />
      </label>
      <label>
        How many cards per person?
        <input
          id='cards'
          type="number"
          name="inputCards"
          value={props.inputCards}
          min="1"
          max="52"
          onChange={(event) => props.onChange(event)} />
      </label>
      <input id='submit' type="submit" value="Play!" />
    </form>
  );
}

export default Form;
