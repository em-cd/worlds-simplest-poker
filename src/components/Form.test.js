import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';

describe('Form', () => {
  it('renders correctly', () => {
    const form = shallow(<Form />);
    expect(form).toMatchSnapshot();
  });
});
