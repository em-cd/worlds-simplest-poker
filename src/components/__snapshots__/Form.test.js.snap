// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Form />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label>
          How many players?
          <input
            id="players"
            max="52"
            min="2"
            name="inputPlayers"
            onChange={[Function]}
            type="number"
          />
        </label>,
        <label>
          How many cards per person?
          <input
            id="cards"
            max="52"
            min="1"
            name="inputCards"
            onChange={[Function]}
            type="number"
          />
        </label>,
        <input
          id="submit"
          type="submit"
          value="Play!"
        />,
      ],
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "How many players?",
            <input
              id="players"
              max="52"
              min="2"
              name="inputPlayers"
              onChange={[Function]}
              type="number"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          "How many players?",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "players",
              "max": "52",
              "min": "2",
              "name": "inputPlayers",
              "onChange": [Function],
              "type": "number",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "How many cards per person?",
            <input
              id="cards"
              max="52"
              min="1"
              name="inputCards"
              onChange={[Function]}
              type="number"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          "How many cards per person?",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "cards",
              "max": "52",
              "min": "1",
              "name": "inputCards",
              "onChange": [Function],
              "type": "number",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "submit",
          "type": "submit",
          "value": "Play!",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label>
            How many players?
            <input
              id="players"
              max="52"
              min="2"
              name="inputPlayers"
              onChange={[Function]}
              type="number"
            />
          </label>,
          <label>
            How many cards per person?
            <input
              id="cards"
              max="52"
              min="1"
              name="inputCards"
              onChange={[Function]}
              type="number"
            />
          </label>,
          <input
            id="submit"
            type="submit"
            value="Play!"
          />,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "How many players?",
              <input
                id="players"
                max="52"
                min="2"
                name="inputPlayers"
                onChange={[Function]}
                type="number"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            "How many players?",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "players",
                "max": "52",
                "min": "2",
                "name": "inputPlayers",
                "onChange": [Function],
                "type": "number",
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "How many cards per person?",
              <input
                id="cards"
                max="52"
                min="1"
                name="inputCards"
                onChange={[Function]}
                type="number"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            "How many cards per person?",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "cards",
                "max": "52",
                "min": "1",
                "name": "inputCards",
                "onChange": [Function],
                "type": "number",
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "submit",
            "type": "submit",
            "value": "Play!",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
